app/src/main/java/com/example/triptracker/view/map/RecordScreen.kt

```diff
@@ -0,0 +1,406 @@
+const val DELAY = 5000L
+
+/**
+ * Composable function for displaying the RecordScreen.
+ *
+ * @param context The application context.
+ * @param viewModel The RecordViewModel instance.
+ */
+@Composable
+fun RecordScreen(context: Context, viewModel: RecordViewModel = RecordViewModel()) {
+  // default device location is EPFL
+  var deviceLocation = LatLng(46.518831258, 6.559331096)
+
+  // Default map properties and UI settings
+  var mapProperties =
+      MapProperties(
+          mapType = MapType.NORMAL, isMyLocationEnabled = checkForLocationPermission(context))
+  var uiSettings = MapUiSettings(myLocationButtonEnabled = checkForLocationPermission(context))
+
+  // Get current device location
+  getCurrentLocation(context = context, onLocationFetched = { deviceLocation = it })
+
+  // Check for location permission
+  when (checkForLocationPermission(context = context)) {
+    true -> {
+      Map(context, viewModel, deviceLocation)
+    }
+    false -> {
+      AllowLocationPermission(
+          onPermissionGranted = {
+            mapProperties = MapProperties(mapType = MapType.NORMAL, isMyLocationEnabled = true)
+            uiSettings = MapUiSettings(myLocationButtonEnabled = true)
+          },
+          onPermissionDenied = {
+            mapProperties = MapProperties(mapType = MapType.NORMAL, isMyLocationEnabled = false)
+            uiSettings = MapUiSettings(myLocationButtonEnabled = false)
+          })
+    }
+  }
+}
+/**
+ * Composable function for displaying the map.
+ *
+ * @param context The application context.
+ * @param viewModel The RecordViewModel instance.
+ * @param startLocation The starting location.
+ */
+@Composable
+fun Map(context: Context, viewModel: RecordViewModel, startLocation: LatLng) {
+  // Mutable state for the device location and the local LatLng list
+  var deviceLocation by remember { mutableStateOf(startLocation) }
+  val localLatLngList = remember { mutableStateListOf<LatLng>() }
+
+  // Coroutine scope for the animations
+  val coroutineScope = rememberCoroutineScope()
+
+  // Remember camera position state
+  val cameraPositionState = rememberCameraPositionState {
+    position = CameraPosition.fromLatLngZoom(deviceLocation, 17f)
+  }
+
+  // Animate camera to device location
+  LaunchedEffect(key1 = deviceLocation) {
+    coroutineScope.launch {
+      cameraPositionState.animate(
+          CameraUpdateFactory.newCameraPosition(CameraPosition.fromLatLngZoom(deviceLocation, 17f)))
+    }
+  }
+
+  // Update device location
+  LaunchedEffect(Unit) {
+    while (true) {
+      getCurrentLocation(context = context, onLocationFetched = { deviceLocation = it })
+      delay(DELAY)
+    }
+  }
+
+  // Add current location to list every "DELAY" seconds if recording
+  LaunchedEffect(key1 = viewModel.isRecording()) {
+    while (viewModel.isRecording()) {
+      if (!viewModel.isPaused.value) {
+        viewModel.addLatLng(deviceLocation)
+        if (viewModel.latLongList.isNotEmpty()) {
```

Comment:
Why do we store on different lists (one in the view model and one in the class itself ?

Reply:
It's the only way to display the path in live tracking that I found. I added comment to explain this